module/zfs/zfs_ioctl.c

```diff
@@ -6297,7 +6301,9 @@ static void
 zfs_allow_log_destroy(void *arg)
 {
 	char *poolname = arg;
-	strfree(poolname);
+	
+	if (poolname != NULL)
+	    strfree(poolname);
```

Comment:
The problem is to be, 
I have add a log to zfs_allow_log_destroy function and zfs_ioc_log_history function. 
void
zfs_allow_log_destroy(void *arg)
{
    char *poolname = arg;
    if(NULL != poolname)
    {
	 printk("heary-cao zfs_allow_log_destroy----------> poolname is %s\n",poolname);
         strfree(poolname);
    }
    else
    {
        printk("heary-cao zfs_allow_log_destroy----------> poolname is NULL\n");
    }
}
zfs_ioc_log_history:
    poolname = tsd_get(zfs_allow_log_key);
    if(poolname == NULL)
    {
		printk("heary-cao zfs_ioc_log_history ------>tsd_get  error\n");
		return (SET_ERROR(EINVAL));
    }
   else
   {
	printk("heary-cao zfs_ioc_log_history--->poolname:%s\n",poolname);
   }

run zfs test suite:
[/opt/ztest/zfs-tests/tests/functional/cli_root/zdb]
tests = ['zdb_001_neg']
[/opt/ztest/zfs-tests/tests/functional/cli_root/zfs]
tests = ['zfs_001_neg', 'zfs_003_neg']
[/opt/ztest/zfs-tests/tests/functional/cli_root/zfs_create]
tests = ['zfs_create_001_pos', 'zfs_create_002_pos', 'zfs_create_003_pos',
         'zfs_create_006_pos', 'zfs_create_007_pos', 'zfs_create_013_pos']

the log is:
[ 2085.673959] heary-cao zfs_ioc_log_history--->poolname:vd-F0703B79-EBB9-5B12-6C74-BAE161A49000
[ 2085.683826] heary-cao zfs_allow_log_destroy----------> poolname is NULL